evmcore/evm.go

```diff
@@ -50,7 +50,7 @@ func NewEVMBlockContext(header *EvmHeader, chain DummyChain, author *common.Addr
 	if header.BaseFee != nil {
 		baseFee = new(big.Int).Set(header.BaseFee)
 	}
-	if header.Difficulty.Sign() == 0 {
+	if header.Difficulty != nil && header.Difficulty.Sign() == 0 {
```

Comment:
@jenikd Does header.Difficulty==nil happens? In which conditions? Are we sure here should not be `header.Difficulty == nil || header.Difficulty.Sign() == 0`?

Reply:
You are right, because I've added difficulty initialization into `ToEvmHeader` function, now in this place `header.Difficulty` will not be nil. Without it it happened, when I wanted to replay any block with debug.